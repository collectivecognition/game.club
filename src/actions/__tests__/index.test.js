import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import * as actions from '../'

jest.mock('../../api')

const mockStore = configureMockStore([thunk])

describe('actions', () => {
  it('should fetch no games', () => {
    const store = mockStore({games: {}})
    return store.dispatch(actions.fetchGames('Zelda'))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      })
  })

  it('should fetch some games', () => {
    const store = mockStore({games: {}})
    return store.dispatch(actions.fetchGames('Halo'))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      })
  })
})