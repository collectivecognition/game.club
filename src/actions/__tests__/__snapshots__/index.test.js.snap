exports[`actions should fetch no games 1`] = `
Array [
  Object {
    "searchTerm": "Zelda",
    "type": "REQUEST_GAMES",
  },
  Object {
    "games": Array [],
    "type": "RECEIVE_GAMES",
  },
]
`;

exports[`actions should fetch some games 1`] = `
Array [
  Object {
    "searchTerm": "Halo",
    "type": "REQUEST_GAMES",
  },
  Object {
    "games": Array [
      Object {
        "name": "Halo",
      },
      Object {
        "name": "Halo 2",
      },
    ],
    "type": "RECEIVE_GAMES",
  },
]
`;
